app/controllers/api/go_cardless_controller.rb

```diff
@@ -45,8 +43,15 @@ def webhook
 
   private
 
+  def builder
+    PaymentProcessor::GoCardless::Transaction
+  end
+
   def success_url
-    local_params = URI.parse(request.url).query
+    local_params =  Rack::Utils.parse_query(
+      URI.parse(request.url).query
+    ).merge( params.slice(:page_id) ).to_query
+
     "#{request.base_url}/api/go_cardless/payment_complete?#{local_params}"
   end
 end
```

Comment:
I see that the director class is still here... just flagging it because I know you had ideas on where it should go and what it should be named.